contrib/linearize/linearize-data.py

```diff
@@ -213,8 +213,8 @@ def run(self):
 
             inMagic = inhdr[:4]
             if (inMagic != self.settings['netmagic']):
-                print("Invalid magic: " + inMagic.hex())
-                return
+                self.inF.seek(-7, os.SEEK_CUR)
```

Comment:
Care to add a comment here?

Reply:
Certainly. When searching for the start of the next block within the block file, 8 bytes are read: first four bytes should consist of the magic bytes and the next four, the length of the block. Rather than quitting if the magic bytes are not found, this patch rewinds the file cursor 7 bytes so the routine can attempt the search again by reading and comparing the next 8 bytes skipping the first byte of the previous search. It continues this process until it finds the magic bytes or reaches the end of the file.

There are rare occasions when there is a gap between blocks within the block file. This patch allows the script to compensate by finding the position to begin reading the next block while skipping the gap.

Reply:
I mean in the code, a brief comment.

Reply:
Right. Just added one now.